service/src/main/java/com/ge/predix/acs/attribute/connector/management/AttributeConnectorServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+import com.ge.predix.acs.rest.AttributeAdapterConnection;
+import com.ge.predix.acs.rest.AttributeConnector;
+import com.ge.predix.acs.zone.management.dao.ZoneEntity;
+import com.ge.predix.acs.zone.management.dao.ZoneRepository;
+import com.ge.predix.acs.zone.resolver.ZoneResolver;
+
+@Component
+public class AttributeConnectorServiceImpl implements AttributeConnectorService {
+    private static final int CACHED_INTERVAL_THRESHOLD_MINUTES = 30;
+
+    @Autowired
+    private ZoneRepository zoneRepository;
+    @Autowired
+    private ZoneResolver zoneResolver;
+
+    private final ConnectorConverter connectorConverter = new ConnectorConverter();
+
+    @Override
+    public boolean upsertResourceConnector(final AttributeConnector connector) {
+        ZoneEntity zoneEntity = this.zoneResolver.getZoneEntityOrFail();
+        validateConnectorConfigOrFail(connector);
+
+        AttributeConnectorEntity connectorEntity = zoneEntity.getResourceAttributeConnector();
+        boolean isCreated = null == connectorEntity;
+        
+        connectorEntity = connectorConverter.toConnectoEntity(connector);
+        zoneEntity.setResourceAttributeConnector(connectorEntity);
+        try {
+            this.zoneRepository.save(zoneEntity);
+        } catch (Exception e) {
+            String message = String.format(
+                    "Unable to persist connector configuration for resource attributes for zone '%s'",
+                    zoneEntity.getName());
+            throw new AttributeConnectorException(message, e);
+        }
+        return isCreated;
+    }
+
+    @Override
+    public AttributeConnector retrieveResourceConnector() {
+        ZoneEntity zoneEntity = this.zoneResolver.getZoneEntityOrFail();
+        return this.connectorConverter.toConnector(zoneEntity.getResourceAttributeConnector());
+    }
+
+    @Override
+    public Boolean deleteResourceConnector() {
+        ZoneEntity zoneEntity = this.zoneResolver.getZoneEntityOrFail();
+        boolean isDeleted = false;
+        if (null == zoneEntity.getResourceAttributeConnector()) {
+            return isDeleted;
+        }
+        zoneEntity.setResourceAttributeConnector(null);
+        try {
+            this.zoneRepository.save(zoneEntity);
+            isDeleted = true;
+        } catch (Exception e) {
+            String message = String.format(
+                    "Unable to delete connector configuration for resource attributes for zone '%s'",
+                    zoneEntity.getName());
+            throw new AttributeConnectorException(message, e);
+        }
+        return isDeleted;
+
+    }
+
+    private void validateAdapterEntityOrFail(final AttributeAdapterConnection adapter) {
+        if (adapter == null) {
+            throw new AttributeConnectorException("Attribute connector configuration requires at least one adapter");
+        }
+        if (adapter.getAdapterEndpoint() == null || adapter.getAdapterEndpoint().isEmpty()) {
```

Comment:
maybe this is what you are referring to earlier - a lot of this validation can be enforced via jackson using annotations on the AttributeAdapterConnection bean.  I believe, by default,  a field is required, unless you mark it as optional.